sqldb/src/config.rs

```diff
@@ -0,0 +1,127 @@
+    root_cert: Option<String>,
+    /// Optional connection pool information
+    #[serde(default)]
+    pool: PoolOptions,
+}
+
+const DEFAULT_MAX_CONNECTIONS: u32 = 8;
+const DEFAULT_MIN_IDLE: u32 = 0;
+const DEFAULT_MAX_LIFETIME_SEC: u32 = 7200;
+const DEFAULT_IDLE_TIMEOUT_SEC: u32 = 600;
+// period of inactivity after which keepalive message is sent to the backend server.
+const DEFAULT_KEEPALIVE_IDLE: Duration = Duration::from_secs(15 * 60);
+
+/// Options for configuring connection pool
+#[derive(Debug, Default, Deserialize)]
+pub(crate) struct PoolOptions {
+    /// sets the maximum number of connections the pool should maintain
+    /// Default: 8
+    max_connections: Option<u32>,
+
+    /// sets the minimum number of idle connections the pool should maintain
+    /// Default: 0 (None)
+    min_idle: Option<u32>,
+
+    /// sets the maximum lifetime of individual connections.
+    /// any connection with a lifetime greater than this will be closed.
+    /// It is usually recommended to retire connections periodiically to
+    /// clean up data structures and caches associated with a session.
+    /// Default: 7200 (2 hours)
+    max_lifetime_secs: Option<u32>,
+
+    /// maximum idle duration for an individual connection.
+    /// Any connection with an idle timeout longer than this will be closed.
+    /// For usage-based database server billing, this can be a cost saver.
+    /// Default: 600 (10 minutes)
+    idle_timeout_secs: Option<u32>,
+}
+
+// Support a variety of configuration possibilities:
+//  'uri' (only) - sets the uri, and uses a default connection pool
+//  'config_json' - json with 'uri' and 'pool' settings
+//  'config_b64' - base64-encoded json wih 'uri' and 'pool' settings
+pub(crate) fn load_config(ld: &LinkDefinition) -> Result<Config, RpcError> {
+    let mut config = Config::default();
+    if let Some(cj) = ld.values.get("config_b64") {
+        config = serde_json::from_slice(
+            &base64::decode(cj)
+                .map_err(|_| RpcError::ProviderInit("invalid config_base64 encoding".into()))?,
+        )
+        .map_err(|e| RpcError::ProviderInit(format!("invalid json config: {}", e)))?;
+    }
+    if let Some(cj) = ld.values.get("config_json") {
+        config = serde_json::from_str(cj.as_str())
+            .map_err(|e| RpcError::ProviderInit(format!("invalid json config: {}", e)))?;
+    }
+    if let Some(uri) = ld.values.get("uri") {
+        config.uri = uri.to_string();
+    }
+    if config.uri.is_empty() {
+        Err(RpcError::ProviderInit(
+            "link params values are missing 'uri'".into(),
+        ))
+    } else {
+        Ok(config)
+    }
+}
+
+pub(crate) async fn create_pool(config: Config) -> Result<crate::Pool, RpcError> {
+    let mut pg_config = tokio_postgres::Config::from_str(&config.uri).map_err(|e| {
+        RpcError::ProviderInit(format!("Invalid db connect string: {}", e.to_string()))
+    })?;
+    pg_config.keepalives_idle(DEFAULT_KEEPALIVE_IDLE);
+    /*
```

Comment:
if we're not intending to support TLS in the initial release, we should delete this comment.

Reply:
intend to support TLS in the initial release. I had coded it but it didn't work yet. Instead of debugging, I commented it out to get a stable build and initial handoff for working on petclinic demo. TLS is a must-have feature.